x/cosmwasmpool/pool_module.go

```diff
@@ -99,12 +102,15 @@ func (k Keeper) GetPool(ctx sdk.Context, poolId uint64) (poolmanagertypes.PoolI,
 func (k Keeper) GetPools(ctx sdk.Context) ([]poolmanagertypes.PoolI, error) {
 	return osmoutils.GatherValuesFromStorePrefix(
 		ctx.KVStore(k.storeKey), types.PoolsKey, func(value []byte) (poolmanagertypes.PoolI, error) {
-			pool := model.Pool{}
+			pool := model.CosmWasmPool{}
 			err := k.cdc.Unmarshal(value, &pool)
 			if err != nil {
 				return nil, err
 			}
-			return &pool, nil
+			return &model.Pool{
+				CosmWasmPool: pool,
+				WasmKeeper:   k.wasmKeeper,
+			}, nil
```

Comment:
I don't follow this change either. We should be careful because this might break CLI serialization.

The reason why we return store model and not the wrapper `model.Pool` is because `model.Pool` is a pure go struct. It cannot be properly serialized or deserialized. As a result, this would break CLI and queries.

For the purposes of import and export genesis, we should be fine operating on `model.CosmwasmPool`. There is no need for `model.Pool`. It is only needed, when we want to run certain `PoolI` methods that require making calls to the contract

Reply:
When removing this change, the following serialization fails, where we need to utilize cosmwasmPool in order to call GetTotalPoolLiquidity

https://github.com/osmosis-labs/osmosis/blob/79b4a8717ec361dcea779ee0360324279e7e806c/x/cosmwasmpool/pool_module.go#L353-L358

Reply:
This change will break the `AllPools` `x/poolmanager` query and CLI because `model.Pool` cannot be serialized.

Please test to make sure. If that's the case, please implement new logic for the `GetTotalPoolLiquidity`

Reply:
@p0mvn should I just make a GetPoolsWasmExtension and only use that when exporting and importing genesis?

Reply:
Yes, that makes sense. However, both `model.Pool` and `model.CosmWasmPool` implement `types.CosmWasmExtension` so the name doesn't seem to convey what we would like. We would like to differentiate between the `model.Pool` that has wasm keeper and `model.CosmWasmPool` that does not.

So I suggest considering `GetPoolsWithWasmKeeper`  or something similar